python/cudf/udf_cpp/shim.cu

```diff
@@ -249,7 +294,7 @@ extern "C" __device__ int string_view_from_udf_string(int* nb_retval,
   return 0;
```

Comment:
I can't comment on the exact line but I'm also becoming concerned about `sv_ptr` not being used here, and how overwriting the constructed object interact with placement `new`.

I presume the destructor for `sv_ptr` would still not be invoked at the end of the scope, or even if it is, it would not be a problem because it wouldn't have a side effect.

I wonder if it's safer to set `nb_retval` to the result of the placement `new`, even though Numba does not look at it. In a similar way to how placement `new` returns a pointer, perhaps the shim functions that invoke it should also be propagating its return value back to the caller.

Should this be (with a `void* nb_retval`):

```c++
nb_retval = new (str) cudf::string_view{*udf_str_ptr};
```

?